StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -170,6 +171,13 @@ public class STPPaymentHandler: NSObject {
         with authenticationContext: STPAuthenticationContext,
         completion: @escaping STPPaymentHandlerActionPaymentIntentCompletionBlock
     ) {
+        let paymentIntentID = paymentParams.stripeId
+        logConfirmPaymentIntentStarted(paymentIntentID: paymentIntentID, paymentParams: paymentParams)
+        // Overwrite completion to send an analytic before calling the caller-supplied completion
+        let completion: STPPaymentHandlerActionPaymentIntentCompletionBlock = { [weak self] status, paymentIntent, error in
+            self?.logConfirmPaymentIntentCompleted(paymentIntentID: paymentIntentID, paymentIntent: paymentIntent, status: status, paymentMethodType: paymentParams.paymentMethodType, error: error)
```

Comment:
Why are we passing in the `paymentIntentID` and the `paymentIntent`, can't we get the ID from the intent itself?

Reply:
The PI is optional; it's nil if we error before/while fetching it